### PR TITLE
feat: support ZodDate type

### PR DESCRIPTION
**Pull request by heivo**
#48

### CODE REVIEW

File: spec/types/date.spec.ts
Comment (georgyangelov) [inline spec/types/date.spec.ts:12]: Since it's a type: string the example should probably be `example: example.toISOString()` instead of the date object itself. This way we're not relying on the yaml or json serializer to decide how to display it for us.

**Review by georgyangelov — COMMENTED** [review]
Nice, thank you!

File: spec/types/date.spec.ts
Comment (heivo) [reply spec/types/date.spec.ts]: Thank you for the suggestion, I'm now serializing examples in the generator to ISO date strings. Is this what you meant?

Comment (heivo) [reply spec/types/date.spec.ts]: Maybe this is too opinionated because not everyone might want to use ISO date strings. Instead we could change the example TS type from Date to string when the schema is a ZodDate (or derive the example type from the openapi metadata type). What do you think?

File: src/openapi-generator.ts
Comment (AGalabov) [inline src/openapi-generator.ts:883]: Hmmm what caused this change? I think that the metadata should override anything coming as a base schema.

File: src/openapi-generator.ts
Comment (AGalabov) [inline src/openapi-generator.ts:865]: Does inferring the type help with anything here? I am okay with this change, just asking if it is cosmetic or if it helped with a TS usecase.

File: src/openapi-generator.ts
Comment (AGalabov) [inline src/openapi-generator.ts:718]: Seeing that you are going with the `toISOString` approach then we could have added a `date-time` `format`. However, looking at [this](https://swagger.io/docs/specification/data-models/data-types/) I see that there are both `date` and `date-time`. This means that for a "user" of `zod-to-openapi` it might be a perfectly valid use-case to use a `date` format. And since the examples are being set by the "user" I'd rather leave this to them to serialize and not touch it at all here.

File: spec/types/date.spec.ts
Comment (AGalabov) [reply spec/types/date.spec.ts]: I think that the example should be defined by the user and not zod-to-openapi. But yes, it should be a string (Typescript wise)

File: src/openapi-generator.ts
Comment (heivo) [reply src/openapi-generator.ts:865]: I added it to have proper types here, otherwise the example would be `any` type.
image

I will probably not need it once I updated the PR but I think it's a good addition nevertheless. But I can also revert it if you want to.

File: spec/types/date.spec.ts
Comment (heivo) [reply spec/types/date.spec.ts]: Ok I changed the example type to `string` when the schema type infers to `Date`. I also reverted the automatically generated example so users can define them.

File: src/zod-extensions.ts
Comment (AGalabov) [inline src/zod-extensions.ts:9]: @heivo can you please pull the latest master. Since in [this PR](https://github.com/asteasolutions/zod-to-openapi/pull/53/files) the same typing has been used for `examples` and `default`. You can also create a reusable type to remove the duplication.

File: spec/types/date.spec.ts
Comment (AGalabov) [inline spec/types/date.spec.ts:20]: No need for `example: example` - they share the same name

**Review by AGalabov — APPROVED** [review]

**Comment by AGalabov** [comment]
@heivo The PR looks good. However the build checks are failing - presumably because of prettier. Please run `npm run lint` (it checks for errors) and `npm run prettier` (fixes them automatically) and finally `npm test` to verify everything is working fine. I assume there are styling changes that should be made. After that I'll merge your PR

**Comment by heivo** [comment]
> @heivo The PR looks good. However the build checks are failing - presumably because of prettier. Please run `npm run lint` (it checks for errors) and `npm run prettier` (fixes them automatically) and finally `npm test` to verify everything is working fine. I assume there are styling changes that should be made. After that I'll merge your PR

The build fails because of a type error but I think it's not related to my changes, the same happens for me in master branch as well

**Comment by AGalabov** [comment]
@heivo you are right. It must have happened when the latest PRs have been merged (especially the one for the upgrade of `openapi3-ts`). I opened a [PR](https://github.com/asteasolutions/zod-to-openapi/pull/56) with the fix that seemed best suited to me. I'll check with @georgyangelov if we can agree on it. Sorry for the inconvenience. Will let you know once a fix is merged.

**Comment by AGalabov** [comment]
Well that was quick :smile: .... @heivo we merged the PR so you should be able to fix the issues once you pull master into your branch.

**Review by georgyangelov — APPROVED** [review]

**Comment by heivo** [comment]
> Well that was quick smile .... @heivo we merged the PR so you should be able to fix the issues once you pull master into your branch.

@AGalabov I merged the master branch

**Pull request merged**
